### PR TITLE
Tp usermodel

### PR DESCRIPTION
**Pull request by capricornication**
Please take a look at the creating of a new user. Boom goes the dynamite!

### CODE REVIEW

File: app/assets/javascripts/users.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (troyleach) [inline app/assets/javascripts/users.coffee:3]: please remove this file

just delete the file - all coffee files

File: app/controllers/auth0_controller.rb [hunk 2]
```diff
@@ -1,16 +1,30 @@
 class Auth0Controller < ApplicationController
   def callback
-    # This stores all the user information that came from Auth0
-    # and the IdP
-
+    
     session[:userinfo] = request.env['omniauth.auth']
+    @user_info = session[:userinfo]
+
```
Comment (troyleach) [inline app/controllers/auth0_controller.rb:6]: why are we using the @ no need for an instance variable

File: app/controllers/auth0_controller.rb [hunk 3]
```diff
@@ -1,16 +1,30 @@
...
+    params[:token] = @user_info.credentials["token"]
+    params[:email] = @user_info.info["email"]
 
-    # Redirect to the URL you want after successfull auth
-    redirect_to '/blogs'
+    users = User.all
+    if users.find_by_email(params[:email]).nil?
+      new_user
+    else
+      redirect_to '/blogs'
```
Comment (troyleach) [inline app/controllers/auth0_controller.rb:14]: please change redirect_to...

```
redirect_to blogs_path
```

File: app/controllers/auth0_controller.rb [hunk 4]
```diff
@@ -1,16 +1,30 @@
...
+    
   end
 
-  def user
-    @user = session[:userinfo]
+  def new_user
+      @user = User.new(pass_info) 
+      if @user.save
+        flash[:success] = "Welcome To Spark Start"
+        redirect_to '/profile/new'
```
Comment (troyleach) [inline app/controllers/auth0_controller.rb:23]: redirect_to profile_new_path

File: app/controllers/auth0_controller.rb [hunk 5]
```diff
@@ -1,16 +1,30 @@
...
+      else
+        flash[:error] = "Try Again"
+        render '/'
```
Comment (troyleach) [inline app/controllers/auth0_controller.rb:26]: render root

File: app/controllers/auth0_controller.rb [hunk 6]
```diff
@@ -1,16 +1,30 @@
...
+    else
+      redirect_to '/blogs'
+    end
+    
   end
 
-  def user
-    @user = session[:userinfo]
+  def new_user
+      @user = User.new(pass_info)
```
Comment (troyleach) [inline app/controllers/auth0_controller.rb:20]: can we please use `user_params`
this is a little more descriptive as to what you are passing

File: app/controllers/auth0_controller.rb [hunk 7]
```diff
@@ -23,4 +37,11 @@ def failure
     # show a failure page or redirect to an error page
     @error_msg = request.params['message']
   end
+
+  private
+
+    def pass_info
```
Comment (troyleach) [inline app/controllers/auth0_controller.rb:43]: please see note above

File: spec/factories/users.rb [hunk 8]
```diff
@@ -1,7 +0,0 @@
-  # FactoryGirl.define do
-  #     factory :user do
-  #       email 'justinviegelahn@gmail.com'
-  #       password 'secret'
-  #       password_confirmation 'secret'
-  #     end
-  # end
```
Comment (troyleach) [inline spec/factories/users.rb:7]: do NOT delete this

Comment (troyleach) [inline spec/factories/users.rb:7]: Why??

File: app/assets/javascripts/users.coffee (hunk 1 above)
Comment (capricornication) [inline app/assets/javascripts/users.coffee]: unable to locate. troy as our witness

Comment (capricornication) [reply app/assets/javascripts/users.coffee:3]: Dunzo!

File: app/controllers/auth0_controller.rb (hunk 2 above)
Comment (capricornication) [reply app/controllers/auth0_controller.rb:6]: good point. no longer an instance variable.

File: app/controllers/auth0_controller.rb (hunk 3 above)
Comment (capricornication) [reply app/controllers/auth0_controller.rb:14]: Dunzo!

File: app/controllers/auth0_controller.rb (hunk 4 above)
Comment (capricornication) [reply app/controllers/auth0_controller.rb:23]: Dunzo!

File: app/controllers/auth0_controller.rb (hunk 5 above)
Comment (capricornication) [reply app/controllers/auth0_controller.rb:26]: Dunzo

File: app/controllers/auth0_controller.rb (hunk 6 above)
Comment (capricornication) [reply app/controllers/auth0_controller.rb:20]: Dunzo

File: app/controllers/auth0_controller.rb (hunk 7 above)
Comment (capricornication) [reply app/controllers/auth0_controller.rb:43]: Dunzo

File: spec/factories/users.rb (hunk 8 above)
Comment (capricornication) [reply spec/factories/users.rb:7]: Sorry. Stefan made us do it.

**Comment by capricornication** [comment]
Updated. Please review.

**Pull request closed without merging**